packages/react/src/primitives/PhoneNumberField/CountryCodeSelect.tsx

```diff
@@ -0,0 +1,29 @@
+import { useMemo } from 'react';
+import { countryDialCodes } from '@aws-amplify/ui';
+
+import { SelectField } from '../SelectField';
+import { ComponentClassNames } from '../shared/constants';
+import { CountryCodeSelectProps } from '../types';
+
+export const CountryCodeSelect: React.FC<CountryCodeSelectProps> = (props) => {
+  const countryCodeOptions = useMemo(
+    () =>
+      countryDialCodes.map((dialCode) => (
+        <option key={dialCode} value={dialCode}>
+          {dialCode}
+        </option>
+      )),
+    countryDialCodes
```

Comment:
dependency array should just have the object reference in it, rather than all the codes.
```suggestion
    [countryDialCodes]
```